Move Scenes list to sidebar

vlaminck opened the pull request:
Closes #508 
Closes #503
Closes #509


# What

1. Move the list of Scenes into a new Scenes tab in sidebar utilizing the `SidebarListItem` framework that was added during the the Tokens Panel unification. 
2. Allow Scenes to be added to folders. (cannot be moved yet)
3. Allow Searching of Scenes
4. Refactored some of the `SidebarListItem` framework to be more generic and reusable.
5. This is only available if `window.CLOUD` is true. Non-migrated users will continue to use the old window at the top of the screen.


# Other Considerations

We discussed not showing images in the sidebar to reduce the initial download and memory impact. I chose to leave it in for now so we could all play with it before making a final decision on it.


# Demos
### Folder and scene creation
![avtt-scenes-panel-demo-1](https://user-images.githubusercontent.com/584771/169044585-af4205b9-8c9b-4ea3-aed6-2e2296e65e19.gif)

### Display larger image on hover
![avtt-flyout-styling](https://user-images.githubusercontent.com/584771/169056811-f2be9a71-acfc-4099-b1f9-7ae91aa635cf.gif)


### Scene and folder deletion
![avtt-scenes-panel-deletion](https://user-images.githubusercontent.com/584771/169046345-dffec206-a4b3-4b34-b564-e218df81c053.gif)

### Scene search
![avtt-scenes-panel-search](https://user-images.githubusercontent.com/584771/169070544-2ae2b4f0-1b99-438f-9a5e-19e1d398ea5f.gif)

### Moving Scenes between folders
https://user-images.githubusercontent.com/584771/169146013-16312f54-9bee-48ac-bcc4-e15684a925b2.mov

bain2236 commented:
couple of things but even still this is soooooooooooooo nice and polished compared to the last dropdown. 
- I would say the image popout is a little fast that I wasn't expecting it :D despite literally seeing a video of it yesterday I completely forgot.
- "New scene" buttons on hover is a copy paste mistake
- when changing scene the red active colour for the players flickers. I throttled my network speed and it does look like when it's loading a new scene it doesn't know where the players are.

vlaminck commented:
> * I would say the image popout is a little fast that I wasn't expecting it :D despite literally seeing a video of it yesterday I completely forgot.

I left this in for now. I thought it would be good to get others playing with it, and then we can remove it before releasing it if we want to.

> * "New scene" buttons on hover is a copy paste mistake

Fixed. good catch!

> * when changing scene the red active colour for the players flickers. I throttled my network speed and it does look like when it's loading a new scene it doesn't know where the players are.

I was blindly removing the class when the DM button was pressed. I changed it to explicitly remove the class from DM buttons only, and now it doesn't flicker.
